libs/SalesforceSDKCore/SalesforceSDKCore/Classes/PushNotification/SFPushNotificationManager.m

```diff
@@ -184,6 +151,15 @@ - (BOOL)registerSalesforceNotificationsWithCompletionBlock:(void (^)(void))compl
         [bodyDict addEntriesFromDictionary: _customPushRegistrationBody];
     }
     
+    if (self.encryptionEnabled) {
+        NSString *rsaPublicKey = [SFSDKPushNotificationEncryption getRSAPublicKey];
+        if (!rsaPublicKey) {
+            [SFSDKCoreLogger e:[self class] format:@"Cannot register for notifications with Salesforce: no RSA key for encrypted notifications"];
+            return NO;
```

Comment:
This is interesting. If we can't get a key, should we fallback on unencrypted push? 🤔

Reply:
Under what circumstances can getRSAPublicKey fail?

Reply:
- I was on the fence about it, my thought was that if an app required encrypted push we shouldn't fall over to something less secure without them knowing
- I don't know what could cause failure, just that the methods this eventually calls into for creation (`SecKeyCreateRandomKey`, `SecKeyCopyPublicKey`) and retrieval are nullable